tests/io/hdf5_test.py

```diff
@@ -84,6 +91,12 @@ def test_variable_eigen():
     check_roundtrip(eigen_2d['x', 0])
 
 
+def test_transform_types():
+    check_roundtrip(rotation)
+    check_roundtrip(translation)
+    check_roundtrip(affine)
```

Comment:
If you compare to `test_variable_eigen()` above, it is tested for 1d as well as 0d variables. Can you extend this to also check 1d? If I remember correctly there was a branch in the implementation that has caused trouble in the past.